tests/integration/longhorn/longhorn_int_test.go

```diff
@@ -0,0 +1,133 @@
+				return testutil.K3sDefaultDeployments()
+			}, "120s", "5s").Should(Succeed())
+		})
+	})
+
+	When("longhorn is installed", func() {
+		It("installs components into the longhorn-system namespace", func() {
+			result, err := testutil.K3sCmd("kubectl apply -f ./testdata/longhorn.yaml")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(ContainSubstring("namespace/longhorn-system created"))
+			Expect(result).To(ContainSubstring("daemonset.apps/longhorn-manager created"))
+			Expect(result).To(ContainSubstring("deployment.apps/longhorn-driver-deployer created"))
+			Expect(result).To(ContainSubstring("deployment.apps/longhorn-recovery-backend created"))
+			Expect(result).To(ContainSubstring("deployment.apps/longhorn-ui created"))
+			Expect(result).To(ContainSubstring("deployment.apps/longhorn-conversion-webhook created"))
+			Expect(result).To(ContainSubstring("deployment.apps/longhorn-admission-webhook created"))
+		})
+		It("starts the longhorn pods with no problems", func() {
+			Eventually(func() error {
+				pods, err := testutil.ParsePods("longhorn-system", metav1.ListOptions{})
+				if err != nil {
+					return err
+				}
+				for _, pod := range pods {
+					if pod.Status.Phase != "Running" && pod.Status.Phase != "Succeeded" {
+						return fmt.Errorf("pod %s failing", pod.Name)
+					}
+				}
+				return nil
+			}, "120s", "5s").Should(Succeed())
+		})
+	})
+
+	When("persistent volume claim is created", func() {
+		It("creates the pv and pvc", func() {
+			result, err := testutil.K3sCmd("kubectl create -f ./testdata/pvc.yaml")
+			Expect(result).To(ContainSubstring("persistentvolumeclaim/longhorn-volv-pvc created"))
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				pvc, err := testutil.GetPersistentVolumeClaim("default", "longhorn-volv-pvc")
+				if err != nil {
+					return fmt.Errorf("failed to get pvc longhorn-volv-pvc")
+				}
+				if pvc.Status.Phase != "Bound" {
+					return fmt.Errorf("pvc longhorn-volv-pvc not bound")
+				}
+				pv, err := testutil.GetPersistentVolume(pvc.Spec.VolumeName)
+				if err != nil {
+					return fmt.Errorf("failed to get pv %s", pvc.Spec.VolumeName)
+				}
+				if pv.Status.Phase != "Bound" {
+					return fmt.Errorf("pv %s not bound", pv.Name)
+				}
+				return nil
+			}, "300s", "5s").Should(Succeed())
+		})
+		It("creates a pod with the pvc", func() {
+			result, err := testutil.K3sCmd("kubectl create -f ./testdata/pod.yaml")
+			Expect(result).To(ContainSubstring("pod/volume-test created"))
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				pod, err := testutil.GetPod("default", "volume-test")
+				if err != nil {
+					return fmt.Errorf("failed to get pod volume-test")
+				}
+				if pod.Status.Phase != "Running" {
+					return fmt.Errorf("pod volume-test \"%s\" reason: \"%s\" message \"%s\"", pod.Status.Phase, pod.Status.Reason, pod.Status.Message)
+				}
+				return nil
+			}, "60s", "5s").Should(Succeed())
+		})
+	})
```

Comment:
Can you add a section deletion the longhorn pvc? Also doubles as cleanup for test.

Reply:
Yep, it looks like this test cleans up the storage replicas.